### PR TITLE
feat(swapper): base unit vernacular

### PR DESCRIPTION
**Pull request by gomesalexandre**
<h3 align="center">Description</h3>

<p align="center">
  
</p>

Plumbing PR bringing the base unit vernacular in many places across lib, making sure we:

1. always name things the correct way, fixing some "precision" named vars that were actually base units
2. always use the vernacular, either for our in-house vars, or using destructuring to re-assign API-returned values to the right vernacular

Note that this is only property name changes (no runtime change), but will require web to accommodate for these (https://github.com/shapeshift/web/pull/3441 as a bigger boy, https://github.com/shapeshift/web/pull/3475 for the time being to make the diff less crazy on the former)

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1122** <a href="https://app.graphite.dev/github/pr/shapeshift/lib/1122" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/lib/1122?utm_source=stack-comment).

File: packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts [hunk 1]
```diff
@@ -130,28 +130,29 @@ const expectedTradeWethToFox: CowTrade<KnownChainIds.EthereumMainnet> = {
...
-      estimatedGas: '100000',
-      gasPrice: '79036500000',
-      approvalFee: '7903650000000000',
+const expectedTradeQuoteWbtcToWethWithapprovalFeeCryptoBaseUnit: CowTrade<KnownChainIds.EthereumMainnet> =
+  {
+    rate: '19.13939810252384532346', // 19.14 WETH per WBTC
+    feeData: {
+      chainSpecific: {
+        estimatedGas: '100000',
+        gasPrice: '79036500000',
```
Comment (0xApotheosis) [inline packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts:139]: Do we want to do `gasPrice`, too?

File: packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts [hunk 2]
```diff
@@ -130,28 +130,29 @@ const expectedTradeWethToFox: CowTrade<KnownChainIds.EthereumMainnet> = {
...
-  buyAmountCryptoPrecision: '19136098853078932263', // 19.13 WETH
-  sources: [{ name: SwapperName.CowSwap, proportion: '1' }],
-  buyAsset: WETH,
-  sellAsset: WBTC,
-  bip44Params: { purpose: 44, coinType: 60, accountNumber: 0 },
-  receiveAddress: 'address11',
-  feeAmountInSellToken: '17238',
-  sellAmountWithoutFee: '99982762',
-}
+    sellAmountCryptoPrecision: '100000000',
```
Comment (0xApotheosis) [inline packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts:146]: This is not crypto precision, it's base unit.

File: packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts [hunk 3]
```diff
@@ -130,28 +130,29 @@ const expectedTradeWethToFox: CowTrade<KnownChainIds.EthereumMainnet> = {
...
+    buyAmountCryptoPrecision: '19136098853078932263', // 19.13 WETH
```
Comment (0xApotheosis) [inline packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts:147]: This is not crypto precision, it's base unit.

File: packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts [hunk 4]
```diff
@@ -130,28 +130,29 @@ const expectedTradeWethToFox: CowTrade<KnownChainIds.EthereumMainnet> = {
...
+    sources: [{ name: SwapperName.CowSwap, proportion: '1' }],
+    buyAsset: WETH,
+    sellAsset: WBTC,
+    bip44Params: { purpose: 44, coinType: 60, accountNumber: 0 },
+    receiveAddress: 'address11',
+    feeAmountInSellToken: '17238',
```
Comment (0xApotheosis) [inline packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts:153]: Do we want to name this properly too?

File: packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts [hunk 5]
```diff
@@ -130,28 +130,29 @@ const expectedTradeWethToFox: CowTrade<KnownChainIds.EthereumMainnet> = {
...
+    sellAmountWithoutFee: '99982762',
```
Comment (0xApotheosis) [inline packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.test.ts:154]: Do we want to name this properly too?

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]

File: packages/swapper/src/swappercli.ts [hunk 6]
```diff
@@ -173,7 +169,7 @@ const main = async (): Promise<void> => {
       chainId: sellAsset.chainId as UtxoSupportedChainIds,
       sellAsset,
       buyAsset,
-      sellAmountCryptoPrecision: sellAmountBase,
+      sellAmountBeforeFeesCryptoBaseUnit: sellAmountBase,
```
Comment (gomesalexandre) [inline packages/swapper/src/swappercli.ts:172]: Might as well

Comment (gomesalexandre) [reply packages/swapper/src/swappercli.ts]: 652b55e5fd0953d8019fb33a5d411b789d482cb8

File: packages/swapper/src/swappercli.ts [hunk 7]
```diff
@@ -190,20 +190,22 @@ const main = async (): Promise<void> => {
     return
   }
 
-  const buyAmount = fromBaseUnit(quote.buyAmountCryptoPrecision || '0', buyAsset.precision)
```
Comment (gomesalexandre) [inline packages/swapper/src/swappercli.ts:193]: What was named `buyAmountCryptoPrecision` was actually a base unit

File: packages/swapper/src/swappers/cow/CowSwapper.test.ts [hunk 8]
```diff
@@ -202,25 +202,25 @@ describe('CowSwapper', () => {
...
-            gasPrice: '79036500000',
+            gasPriceCryptoBaseUnit: '79036500000',
           },
           buyAssetTradeFeeUsd: '0',
           sellAssetTradeFeeUsd: '0',
-          networkFee: '14557942658757988',
+          networkFeeCryptoBaseUnit: '14557942658757988',
         },
-        sellAmountWithoutFee: '985442057341242012',
+        sellAmountDeductFeeCryptoBaseUnit: '985442057341242012',
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/cow/CowSwapper.test.ts:223]: Little name change to make things clearer, does "without fee" means the base sell amount without fee deducted, or the base amount minus the fees deducted? Turns out it was the latter, so might as well rename it

File: packages/swapper/src/swappers/cow/cowApprovalNeeded/cowApprovalNeeded.test.ts [hunk 9]
```diff
@@ -53,7 +53,7 @@ describe('cowApprovalNeeded', () => {
     const input: ApprovalNeededInput<KnownChainIds.EthereumMainnet> = {
       quote: {
         ...tradeQuote,
-        sellAmountCryptoPrecision: '10',
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/cow/cowApprovalNeeded/cowApprovalNeeded.test.ts:56]: That was also a base unit

**Review by 0xApotheosis — APPROVED** [review]
LGTM ser 🐌

**Pull request merged**